pkg/prometheus/agent/statefulset.go

```diff
@@ -155,48 +154,19 @@ func makeStatefulSetSpec(
 ) (*appsv1.StatefulSetSpec, error) {
 	cpf := p.GetCommonPrometheusFields()
 
-	pImagePath, err := operator.BuildImagePath(
-		operator.StringPtrValOrDefault(cpf.Image, ""),
-		operator.StringValOrDefault("", c.PrometheusDefaultBaseImage),
-		operator.StringValOrDefault(cpf.Version, operator.DefaultPrometheusVersion),
-		"",
-		"",
-	)
+	pImagePath, err := buildImagePath(cpf, c)
 	if err != nil {
 		return nil, err
 	}
 
 	cpf.EnableFeatures = append(cpf.EnableFeatures, "agent")
-	promArgs := prompkg.BuildCommonPrometheusArgs(cpf, cg)
-	promArgs = appendAgentArgs(promArgs, cg, cpf.WALCompression)
-
-	var ports []v1.ContainerPort
-	if !cpf.ListenLocal {
-		ports = []v1.ContainerPort{
-			{
-				Name:          cpf.PortName,
-				ContainerPort: 9090,
-				Protocol:      v1.ProtocolTCP,
-			},
-		}
-	}
+	promArgs := buildAgentArgs(cpf, cg)
 
-	volumes, promVolumeMounts, err := prompkg.BuildCommonVolumes(p, tlsSecrets)
+	volumes, promVolumeMounts, configReloaderVolumeMounts, err := prompkg.BuildVolumes(p, tlsSecrets)
```

Comment:
I'd be fine with keeping BuildCommonVolumes() here and having another call to `CreateConfigReloaderVolumeMounts() like in the server part.